docs/history/evolution-of-design-time-assemblies.md

```diff
@@ -0,0 +1,53 @@
+
+## Definitions
+- **Type Forward:** A type forward is an entry in an Assembly Exported Type Table which contains a type reference to a type in another assembly. It is used if you want to move a type from one assembly to another and still allow for pre-existing assemblies that reference that type to have it resolved at runtime.
+- **Implementation Assembly:** The raw standard IL assembly which contains all the IL code that needs to be executed at runtime.
+- **Reference Assembly:** This is a standard IL assembly but it only contains metadata and no IL code. In other words only the declarations and no definitions, you can think of this similar to a C++ header file.
+- **Contract Assembly:** This is a reference assembly which has been designed to contain a specific set of APIs only. Before .NET Core the APIs in a reference assembly matched 1-to-1 with the corresponding implementation assembly. Contracts are by design not that way, instead we pick and choose the APIs we want to expose and then make the implementation support them. Moreover, in the past reference assemblies had an assembly identity matching that of the corresponding implementation assembly. Contracts break away from that relationship. Simply observing a Contract Assembly does not reveal the relation to the implementation assembly/-ies that may be available at runtime.
+- **Facade Assembly:** This is a standard IL assembly which contains only type forwards. Since type forwards are stored in the exported type table this assembly doesn’t contain any IL and minimal metadata (only the assembly level metadata).
+- **Partial Facade Assembly:** This is similar to a Facade Assembly but it contains a mixture of type forwards as well as some type definitions.
+- **Design time Assembly:** An assembly that is used by tools on a developer machine. Such tools typically include compilers (C#, VB, as well as XAML) and various msbuild and VS tools. They generally don’t need to contain any implementation because they are primarily an abstraction for what a given platform supports.
+- **Runtime Assembly:** This is the raw assembly used by the runtime on an end user’s machine. These are consumed by the runtime (aka CLR/CoreCLR) and need actual code in them to execute.
```

Comment:
I think it'd be good to separate the last two items from the rest a bit, since those are meta items.

E.g. a design time assembly can either be an
1) implementation assembly
2) reference assembly or
3) contract assembly

Similarly, a runtime assembly can either be an
1) implementation assembly
2) partial-facade assembly or
3) facade assembly

That'd make it less confusing in my opinion.